src/snowflake/snowpark/modin/plugin/extensions/timedelta_index.py

```diff
@@ -407,17 +412,39 @@ def mean(
 
         Returns
         -------
-            scalar Timestamp
+            scalar Timedelta
+
+        Examples
+        --------
+        >>> idx = pd.to_timedelta([1, 2, 3, 1], unit='D')
+        >>> idx
+        TimedeltaIndex(['1 days', '2 days', '3 days', '1 days'], dtype='timedelta64[ns]', freq=None)
+        >>> idx.mean()
+        Timedelta('1 days 18:00:00')
 
         See Also
         --------
         numpy.ndarray.mean : Returns the average of array elements along a given axis.
         Series.mean : Return the mean value in a Series.
-
-        Notes
-        -----
-        mean is only defined for Datetime and Timedelta dtypes, not for Period.
         """
+        if axis:
+            # Native pandas raises IndexError: tuple index out of range
+            # We raise a different more user-friendly error message.
+            raise ValueError(
+                f"axis should be 0 for TimedeltaIndex.mean, found '{axis}'"
+            )
+        frame = self._query_compiler._modin_frame
```

Comment:
why we don't have a QC method to do so? Or alternatively, you can do something similar to `to_series()` and `mean` for this

Reply:
I think `agg` is still in-progress by @sfc-gh-mvashishtha, but should be similar?

Reply:
> why we don't have a QC method to do so?

In my understanding we don't have such QC method to operate on index columns, only on data columns. @sfc-gh-yzou would have more context why we don't have that?

> you can do something similar to to_series() and mean for this

We don't have that support yet. It throws NotImplementedError but It would definitely makes sense to reuse code once that is available.

Reply:
Please add a todo link to https://snowflakecomputing.atlassian.net/browse/SNOW-1620439 and we can refactor this code later.

Reply:
done